cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaAgentClient.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.cluster.operator.resource;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.security.GeneralSecurityException;
+import java.security.KeyStore;
+
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManagerFactory;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.fabric8.kubernetes.api.model.Secret;
+import io.strimzi.api.kafka.model.KafkaResources;
+import io.strimzi.operator.cluster.model.DnsNameGenerator;
+import io.strimzi.operator.common.Reconciliation;
+import io.strimzi.operator.common.ReconciliationLogger;
+import io.strimzi.operator.common.Util;
+
+/**
+ * Creates HTTP client and interacts with Kafka Agent's REST endpoint
+ */
+class KafkaAgentClient {
+
+    private static final ReconciliationLogger LOGGER = ReconciliationLogger.create(KafkaAgentClient.class.getName());
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+
+    private static final String BROKER_STATE_REST_PATH = "/v1/broker-state/";
+    private static final int BROKER_STATE_HTTPS_PORT = 8443;
+
+    private String namespace;
+    private Reconciliation reconciliation;
+    private String cluster;
+    private Secret clusterCaCertSecret;
+    private Secret coKeySecret;
+    private HttpClient httpClient;
+
+    KafkaAgentClient(Reconciliation reconciliation, String cluster, String namespace, Secret clusterCaCertSecret, Secret coKeySecret) {
+        this.reconciliation = reconciliation;
+        this.cluster = cluster;
+        this.namespace = namespace;
+        this.clusterCaCertSecret = clusterCaCertSecret;
+        this.coKeySecret = coKeySecret;
+    }
+
+    private HttpClient createHttpClient() throws GeneralSecurityException, IOException {
+
+        SSLContext sslContext = SSLContext.getInstance("TLSv1.3");
+        KeyStore trustStore = KeyStore.getInstance("PKCS12");
+        trustStore.load(new ByteArrayInputStream(
+                        Util.decodeFromSecret(clusterCaCertSecret, "ca.p12")),
+                new String(Util.decodeFromSecret(clusterCaCertSecret, "ca.password"), StandardCharsets.UTF_8).toCharArray());
```

Comment:
Can we use the PEM certificates here in some easy way:?

Reply:
Yes I think we can use the PEM certificates instead. Out of curiosity, can I ask why we prefer to use PEM certificates?

Reply:
It is easier to use and more Kubernetes-friendly format. So using them might be easier for future work around certificate management etc.

Reply:
I mean ... if it would be something super complicated, you should not do it in your PR, we can deal with it later. But if it works easily, it would be nice to use it.

Reply:
As you suggested, I will do this in a separate PR because I would have to make quite few changes.

Reply:
Ok, can you open an issue for it please so that it is not forgotten?

Reply:
created here https://github.com/strimzi/strimzi-kafka-operator/issues/8721